custom_components/zwave_mqtt/entity.py

```diff
@@ -153,12 +153,23 @@ async def async_added_to_hass(self):
     def device_info(self):
         """Return device information for the device registry."""
         node = self.values.primary.node
-        return {
-            "identifiers": {(DOMAIN, node.node_id)},
-            "name": f"{node.node_manufacturer_name} {node.node_product_name}",
-            "manufacturer": node.node_manufacturer_name,
-            "model": node.node_product_name,
-        }
+        instance = self.values.primary.instance
+        # split up device with multiple instances into virtual devices for each instance
+        if instance > 1:
+            return {
+                "identifiers": {(DOMAIN, node.node_id, instance)},
+                "name": f"{node.node_manufacturer_name} {node.node_product_name} - Instance {instance}",
+                "manufacturer": node.node_manufacturer_name,
+                "model": node.node_product_name,
+                "via_device": (DOMAIN, node.node_id),
+            }
+        else:
+            return {
```

Comment:
We can create this dict first and if there are more than one instance modify the needed items in the dict before returning it.

Reply:
Better indeed, I've just adjusted that.